store/tikv/test_batch.go

```diff
@@ -0,0 +1,191 @@
+package tikv
+
+import (
+	"context"
+	"io"
+	"math/rand"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/kvproto/pkg/tikvpb"
+	"github.com/pingcap/tidb/config"
+	"github.com/pingcap/tidb/store/tikv/tikvrpc"
+	"github.com/pingcap/tidb/util/logutil"
+	"go.uber.org/zap"
+	grpcCodes "google.golang.org/grpc/codes"
+	grpcStatus "google.golang.org/grpc/status"
+)
+
+// batchClientTester is a thin wrap over the rpcClient used to test command batching
+type batchClientTester struct {
+	rpcClient Client
+	cfg       BatchClientTestConfig
+	failed    uint32
+	ended     uint32
+	wait      sync.WaitGroup
+}
+
+// BatchClientTestConfig is used to config batchClientTester
+type BatchClientTestConfig struct {
+	Concurrent uint64
+	Timeout    time.Duration
+	MinDelay   uint64
+	MaxDelay   uint64
+	TestLength time.Duration
+}
+
+func (cfg *BatchClientTestConfig) genDelay() uint64 {
+	r := rand.Int63n(int64(cfg.MaxDelay - cfg.MinDelay))
+	return uint64(r) + cfg.MinDelay
+}
+
+// BatchTest start sending test messages to TiKV servers
+func BatchTest(security config.Security, addrs []string, config BatchClientTestConfig) bool {
```

Comment:
Please rename it to ClientTest, because in future it could be not only for batch commands interface.